core/http-auth-aws-crt/src/main/java/software/amazon/awssdk/http/auth/aws/crt/internal/signer/V4aContext.java

```diff
@@ -16,20 +16,33 @@
 package software.amazon.awssdk.http.auth.aws.crt.internal.signer;
 
 import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.crt.auth.signing.AwsSigningConfig;
 import software.amazon.awssdk.http.SdkHttpRequest;
 
 /**
  * A container for data produced during and as a result of the SigV4a request signing with CRT.
  */
 @SdkInternalApi
 public final class V4aContext {
-    private final SdkHttpRequest signedRequest;
+    private final SdkHttpRequest.Builder signedRequest;
+    private final byte[] signature;
+    private final AwsSigningConfig signingConfig;
 
-    public V4aContext(SdkHttpRequest signedRequest) {
+    public V4aContext(SdkHttpRequest.Builder signedRequest, byte[] signature, AwsSigningConfig signingConfig) {
```

Comment:
Is there any reason we changed it to the builder instance? Is this just for testing?

Reply:
No, this is changed because the payload signing step needs access to the request-builder from the context in order to update the request-headers (such as removing content-length, etc.)